ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -38,7 +72,37 @@ def start(self) -> dict:
             ([dict]): Return dictionary. {"status": "", "msg":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("Cluster start operation is not implemented.")
+        if self._is_pcs_cluster_running() is False:
+            _res = self._pcs_cluster_start()
+            if _res != const.STATUSES.SUCCEEDED.value:
+                return {"status": const.STATUSES.FAILED.value, "msg": "Cluster start operation failed"}
+
+        _res = self.node_list()
+        _res = json.loads(_res)
+
+        if _res.get("status") == const.STATUSES.SUCCEEDED.value:
+            _node_list = _res.get("msg")
+            Log.info(f"Node List : {_node_list}")
+            if _node_list is not None:
+                _node_group = [ _node_list[i:i + const.PCS_NODE_START_GROUP_SIZE] for i in range(0, len(_node_list), const.PCS_NODE_START_GROUP_SIZE)]
+
+                for _node_subgroup in _node_group:
+                    for _node_id in _node_subgroup:
+                        _res = self._controllers[const.NODE_CONTROLLER].start(_node_id)
+                        _res = json.loads(_res)
+                        if _res.get("status") == const.STATUSES.FAILED.value:
+                            msg = _res.get("msg")
+                            Log.error(f"Node {_node_id} : {msg}")
+                    # Wait till all the resources get started in the sub group
+                    time.sleep(const.BASE_WAIT_TIME * const.PCS_NODE_START_GROUP_SIZE)
```

Comment:
So we will wait for base time * no. of nodes. But nodes in sub-group will be started parallel, right?

Reply:
Yes

Reply:
These nodes are part of the cluster. While starting the services, they are exchanging messages and this is the reason starting multiple nodes in parallel takes time. This needs to be tuned. Currently, we will use this formula and later we will see if refinement is required.

Reply:
got it!